File: download/src/main/java/com/huison/download/utils/LogUtil.java
package com.huison.download.utils;

import android.util.Log;

/**
 * Created by huisonma on 2019/5/15.
 */
public class LogUtil {

    private static final String TAG = "LogUtil";

    private static boolean sEnable = false;

    public static void enable() {
        sEnable = true;
    }

    public static void d(String msg) {
        d(TAG, msg);
    }

    public static void d(String tag, String msg) {
        if (sEnable) {
            Log.d(tag, msg);
        }
    }

    public static void d(String tag, String msg, Throwable e) {
        if (sEnable) {
            Log.d(tag, msg, e);
        }
    }

    public static void e(String msg) {
        e(TAG, msg);
    }

    public static void e(String tag, String msg) {
        if (sEnable) {
            Log.e(tag, msg);
        }
    }

    public static void e(String tag, String msg, Throwable e) {
        if (sEnable) {
            Log.e(tag, msg, e);
        }
    }
}

File: ktdownload/src/main/java/com/huison/kotlin/download/utils/LogUtil.kt
package com.huison.kotlin.download.utils

import android.util.Log

/**
 * Created by huisonma 2019/9/3.
 */
class LogUtil {

    companion object {

        private const val TAG = "LogUtil"

        private var sEnable = false

        @JvmStatic
        fun enable() {
            sEnable = true
        }

        @JvmStatic
        fun d(msg: String) {
            d(TAG, msg)
        }

        @JvmStatic
        fun d(tag: String, msg: String) {
            if (sEnable) {
                Log.d(tag, msg)
            }
        }

        @JvmStatic
        fun d(tag: String, msg: String, tr: Throwable) {
            if (sEnable) {
                Log.d(tag, msg, tr)
            }
        }

        @JvmStatic
        fun e(msg: String) {
            e(TAG, msg)
        }

        @JvmStatic
        fun e(tag: String, msg: String) {
            if (sEnable) {
                Log.e(tag, msg)
            }
        }

        @JvmStatic
        fun e(tag: String, msg: String, tr: Throwable) {
            if (sEnable) {
                Log.e(tag, msg, tr)
            }
        }
    }
}